code/API_definitions/device-status.yaml

```diff
@@ -205,6 +454,210 @@ components:
+        - ROAMING_CHANGE_COUNTRY
+    DeviceStatusEventTypeList:
+      description: List of Device Status Event Type
+      type: array
+      items:
+        $ref: '#/components/schemas/DeviceStatusEventType'
+      example: [ROAMING_STATUS]
+    CreateSubscription:
+      description: The request for creating a Device Status Subscription
+      type: object
+      required:
+        - webhook
+      properties:
+        subscriptionDetail:
+          $ref: '#/components/schemas/SubscriptionDetail'
+        subscriptionExpireTime:
+          type: string
+          format: date-time
+          example: 2023-01-17T13:18:23.682Z
+          description: The subscription expiration time in date-time format.
+        webhook:
+          $ref: '#/components/schemas/Webhook'
+    Webhook:
+      description: Webhook information for event channel
+      type: object
+      required:
+        - notificationUrl
+      properties: 
+        notificationUrl:
+          type: string
+          example: "https://application-server.com"
+          description: Allows asynchronous delivery of device status related events
+        notificationAuthToken:
+          type: string
+          example: "c8974e592c2fa383d4a3960714"
+          description: Authentification token for callback API  
+    SubscriptionDetail:
+      description: The detail of the requested description
+      type: object
+      required:
+        - eventType
+      properties:
+        device:
+          $ref: '#/components/schemas/Device'
+        eventType:
+          $ref: '#/components/schemas/RoamingEventType'
+    RoamingEventType:
+      type: string
+      description: |
+        ROAMING_STATUS - Device status event triggered when the UE switch from roaming ON to roaming OFF and conversely
+
+        ROAMING_ON - Device status event triggered when the UE switch from roaming OFF to roaming on
+
+        ROAMING_OFF: Device status event triggered when the UE switch from roaming ON to roaming OFF
+
+        ROAMING_CHANGE_COUNTRY: Device status event triggered when the UE in roaming change country code|
+      enum:
+        - ROAMING_STATUS
+        - ROAMING_ON
+        - ROAMING_OFF
+        - ROAMING_CHANGE_COUNTRY
+    CreateNotificationSubscription:
+      allOf:
+        - $ref: '#/components/schemas/CreateSubscription'
+        - type: object
+    SubscriptionInfo:
+      description: Represents a device status subscription.
+      type: object
+      allOf:
+        - $ref: '#/components/schemas/CreateSubscription'
+        - type: object
+          properties:
+            subscriptionId:
+              $ref: '#/components/schemas/SubscriptionId'
+            startsAt:
+              type: string
+              format: date-time
+              description: date time when subscription started
+            expiresAt:
+              type: string
+              format: date-time
+              description: date time when subscription will expire or expired
```

Comment:
Not sure to understand the comment. `subscriptionExpireTime` is the expiration date wished by the requester. Suppose you want to have one month subscription. 
for GET I'm not sure we should enforce  this rule. As subscription will be monetized should we keep 'for some time' before to delete them?

Reply:
From our discussion, I change `expiresAt `to `date time when subscription expired`

Reply:
No, I think it should be "date time when subscription will expire". We are creating a subscriptionand the response says, this subscription starts at "x time" will expire in "y time", aren't we?

Reply:
Not sure we're aligned....

The requester can provide a date wished for subscription: `subscriptionExpireTime `; fine for that.

July 5th, 2023 requester asks for a subscription till July 25,2023  - in the subscription both `subscriptionExpireTime  `& `expireDate `will be valued to July 25th.

Subscription server could also fill a future expire date. For example a subscription server did no support subscription for more the 3 months. If today is July 5th, & requester did no provided `subscriptionExpireTime  `or `subscriptionExpireTime`>3 month from now, then the `expireDate `will be October 5th. 

I assume at this point we're aligned :)

Now, what are we expecting once the `expireDate ` is reached?
Do we assume that the subscription resource is deleted and not anymore accessible by `GET`? or does this resource is still accessible for Tracking for example? I'm not sure about an unified pattern and that's why at last I prefer the initial wording: Date time when subscription expired or will expire.

Reply:
Additionally, I have a basic question. In the case that a subscription has been deleted, will it still be included in the GET eventSubscription? If it is not included, it would restrict the client from accessing the subscription's history. Conversely, if it is included, we should specify whether a specific subscription has been deleted, even if its expiration time is still valid.

Reply:
@sachinvodafone
I had anyway been assuming that `GET /event-subscriptions` would only return active subscriptions. What is the history that you think the API consumer might still need access to for an expired subscription, remembering that CAMARA APIs are designed to be called by code and not by people?

Providing details of expired subscriptions can anyway be implementation specific. But you would need a good argument if filtering options are to be included in the API design.

Reply:
Hi @eric-murray Thanks for your notes. I was referring the notes between @bigludo7 and @monamok on the same topic so thought to clarify if we have reached to any alignment or not. 

Regarding whether subscription history (subscription ID , creation date, expire date, deletion status) is required or not, while not all API consumers may require access to historical data of expired subscriptions, there are scenarios where it can still be relevant and valuable for client.  For example: 

1. In **Audit and Compliance** where  it may be necessary to maintain a complete audit trail of all subscriptions, including expired/deleted ones.
2. In **Billing and Financial Records** where  access to these  data may be valuable for accounting, billing reconciliation, or financial reporting purposes.
3. In **Analysis and Reporting** where historical data on expired subscriptions may be useful for analytical purposes

Reply:
The current API definition would allow `GET /event-subscriptions` to return expired subscriptions, but I don't see that any of the reasons you list would make this mandatory. API consumers can keep their own records of expired subscriptions (there is not that much data per subscription), and the API provider would anyway need to keep records for the reasons you list, but that does not mean they need to make them available to the API consumer through the CAMARA service API.

So I still don't see any reason to mandate that `GET /event-subscriptions` returns expired subscriptions, and hence no reason to add filtering options.

Reply:
If we are sharing only active subscriptions, the description of the "ExpireAt" field should indicate the "date and time when the subscription will expire" instead of mentioning both "when the subscription will expire **or has expired**."

Reply:
I already said that API providers can give details of expired subscriptions if they choose to implement that. But there is no reason to mandate it, nor to prohibit it.

API consumers should not rely on the API for historical records of their subscriptions. They need to be more organised than that.